src/types/instantsearch.ts

```diff
@@ -119,10 +120,19 @@ export type DerivedHelper = EventEmitter & {
 };
 
 export type InstantSearch = {
-  templatesConfig: object;
-  insightsClient: AlgoliaInsightsClient | null;
   helper: Helper | null;
-  widgets: Widget[];
+  widgets: Widget[]; // @TODO: remove once the RoutingManger uses the index
+  mainHelper: Helper | null;
+  mainIndex: Index;
+  insightsClient: AlgoliaInsightsClient | null;
+  templatesConfig: object;
+  _isSearchStalled: boolean;
+  _searchParameters: Partial<SearchParameters>;
+  _createAbsoluteURL(state: Partial<SearchParameters>): string;
+  _onHistoryChange(callback: (state: Partial<SearchParameters>) => void): void;
```

Comment:
Why did we add `_onHistoryChange` back?

Reply:
AFAIC we never removed it, neither on [`develop`](https://github.com/algolia/instantsearch.js/blob/develop/src/lib/InstantSearch.js#L179) nor on [`next`](https://github.com/algolia/instantsearch.js/blob/0a21759db095ad912de9ce1fbada5e6b7d786927/src/lib/InstantSearch.js#L184). We can drop it inside its own PR.